sdg/components.py

```diff
@@ -273,9 +295,17 @@ def get_git_host(repo_url):
         endpoint = os.getenv("endpoint")
     else:
         print("SDG Teacher secret specified, fetching...")
-        api_key, model_name, endpoint = fetch_secret(
-            sdg_secret_name, ["api_token", "model_name", "endpoint"]
+        secret = fetch_secret(
+            sdg_secret_name,
+        )
+        secret_data = secret.get("data", {})
+        api_key = (
+            base64.b64decode(secret_data["api_token"]).decode()
+            if "api_token" in secret_data
+            else ""
         )
+        model_name = base64.b64decode(secret_data["model_name"]).decode()
+        endpoint = base64.b64decode(secret_data["endpoint"]).decode()
         if endpoint is None or model_name is None:
```

Comment:
This will throw a KeyError if the user made a mistake on the secret. How about this?
```suggestion
        model_name = base64.b64decode(secret_data.get("model_name", "")).decode()
        endpoint = base64.b64decode(secret_data.get("endpoint", "")).decode()
        if not endpoint or not model_name:
```